src/ControlSystem/Systems/Size.hpp

```diff
@@ -0,0 +1,194 @@
+                                  control_system::protocols::ControlError>);
+
+  // tag goes in control component
+  struct MeasurementQueue : db::SimpleTag {
+    using type = LinkedMessageQueue<
+        double,
+        tmpl::list<StrahlkorperTags::Strahlkorper<Frame::Distorted>,
+                   QueueTags::ExcisionSurface<Frame::Distorted>,
+                   ::Tags::dt<StrahlkorperTags::Strahlkorper<Frame::Distorted>>,
+                   QueueTags::LapseOnExcisionSurface,
+                   QueueTags::ShiftyQuantity<Frame::Distorted>,
+                   QueueTags::SpatialMetricOnExcisionSurface<Frame::Distorted>,
+                   QueueTags::InverseSpatialMetricOnExcisionSurface<
+                       Frame::Distorted>>>;
+  };
+
+  using simple_tags = tmpl::list<MeasurementQueue>;
+
+  struct process_measurement {
+    template <typename Submeasurement>
+    using argument_tags = tmpl::conditional_t<
+        std::is_same_v<Submeasurement,
+                       typename measurements::CharSpeed<Horizon>::Excision>,
+        tmpl::list<
+            StrahlkorperTags::Strahlkorper<Frame::Grid>,
+            gr::Tags::Lapse<DataVector>,
+            gr::Tags::ShiftyQuantity<DataVector, 3, Frame::Distorted>,
+            gr::Tags::SpatialMetric<DataVector, 3, Frame::Distorted>,
+            gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Distorted>>,
+        tmpl::list<StrahlkorperTags::Strahlkorper<Frame::Distorted>,
+                   ::ah::Tags::TimeDerivStrahlkorper<Frame::Distorted>>>;
+
+    template <typename Metavariables>
+    static void apply(
+        typename measurements::CharSpeed<Horizon>::Excision /*meta*/,
+        const Strahlkorper<Frame::Grid>& grid_excision_surface,
+        const Scalar<DataVector>& lapse,
+        const tnsr::I<DataVector, 3, Frame::Distorted>& shifty_quantity,
+        const tnsr::ii<DataVector, 3, Frame::Distorted>&
+            spatial_metric_on_excision_surface,
+        const tnsr::II<DataVector, 3, Frame::Distorted>&
+            inverse_spatial_metric_on_excision_surface,
+        Parallel::GlobalCache<Metavariables>& cache,
+        const LinkedMessageId<double>& measurement_id) {
+      auto& control_sys_proxy = Parallel::get_parallel_component<
+          ControlComponent<Metavariables, Size>>(cache);
+
+      if (Parallel::get<Tags::Verbosity>(cache) >= ::Verbosity::Verbose) {
+        Parallel::printf("%s, time = %s: Received excision measurement\n",
+                         name(), measurement_id);
+      }
+
+      Strahlkorper<Frame::Distorted> distorted_excision_surface{};
+      strahlkorper_in_different_frame_aligned(
+          make_not_null(&distorted_excision_surface), grid_excision_surface,
+          Parallel::get<domain::Tags::Domain<3>>(cache),
+          Parallel::get<domain::Tags::FunctionsOfTime>(cache),
+          measurement_id.id);
+
+      // excision strahlkorper
+      Parallel::simple_action<::Actions::UpdateMessageQueue<
```

Comment:
Why are there so many messages sent? Can't the data be sent in one message?

Reply:
It didn't occur to me that these were all triggering separate messages.  While that's not as critical for control systems, you're right, they should be merged.

Reply:
The `UpdateMessageQueue` action currently only takes one queue tag at a time. This can of course be changed, but I'll do that in a separate PR.

Reply:
I was thinking just storing a tuple.

Reply:
@wthrowe are you requesting changes for this PR?

Reply:
Instead of having five separate entries in the queue, have one that stores a tagged tuple or a struct or something.